addons/survey/models/survey.py

```diff
@@ -0,0 +1,1077 @@
+    }
+
+    # Actions
+
+    @api.multi
+    def action_start_survey(self):
+        ''' Open the website page with the survey form '''
+        self.ensure_one()
+        trail = ""
+        context = dict(self.env.context, relative_url=True)
+        if 'survey_token' in context:
+            trail = "/" + context['survey_token']
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Start Survey",
+            'target': 'self',
+            'url': self.with_context(context).public_url + trail
+        }
+
+    @api.multi
+    def action_send_survey(self):
+        ''' Open a window to compose an email, pre-filled with the survey
+        message '''
+        self.ensure_one()
+        if not self.designed:
+            raise UserError(_('You cannot send an invitation for a survey that has no questions.'))
+
+        if self.stage_id.closed:
+            raise UserError(_("You cannot send invitations for closed surveys."))
+        template_id = self.env.ref('survey.email_template_survey').id
+        ctx = dict(self.env.context)
+
+        ctx.update({'default_model': 'survey.survey',
+                    'default_res_id': self.id,
+                    'default_survey_id': self.id,
+                    'default_use_template': bool(template_id),
+                    'default_template_id': template_id,
+                    'default_composition_mode': 'comment'}
+                   )
+        return {
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'survey.mail.compose.message',
+            'target': 'new',
+            'context': ctx,
+        }
+
+    @api.multi
+    def action_print_survey(self):
+        ''' Open the website page with the survey printable view '''
+        self.ensure_one()
+        trail = ""
+        context = dict(self.env.context or {}, relative_url=True)
+        if 'survey_token' in context:
+            trail = "/" + context['survey_token']
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Print Survey",
+            'target': 'self',
+            'url': self.with_context(context).print_url + trail
+        }
+
+    @api.multi
+    def action_result_survey(self):
+        ''' Open the website page with the survey results view '''
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Results of the Survey",
+            'target': 'self',
+            'url': self.with_context(relative_url=True).result_url
+        }
+
+    @api.multi
+    def action_test_survey(self):
+        ''' Open the website page with the survey form into test mode'''
+        context = dict(self.env.context or {}, relative_url=True)
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Results of the Survey",
+            'target': 'self',
+            'url': self.with_context(context).public_url + "/phantom"
```

Comment:
self.with_context(relative_url=True).public_url and remove context from line 252
also ensure_one make sense here